scripts/safari.js

```diff
@@ -6828,7 +6804,339 @@ function Safari() {
+                            safaribot.sendMessage(id, "You received " + plural(amt, reward.name, null, true) + " for winning the event!", safchan);
+                        }
+                        if (reward.type == "poke") {
+                            player.pokemon.push(reward.id);
+                        } else {
+                            rewardCapCheck(player, reward.id, amt);
+                        }
+                        safari.saveGame(player);
+                    }
+                }
+            }
+        }
+        this.finished = true;
+    };
+    FactionWar.prototype.join = function(src) {
+        if (this.turn >= 6) {
+            safaribot.sendMessage(src, "The " + this.eventName + " is already underway, you can't join now!", safchan);
+            return;
+        }
+        var player = getAvatar(src);
+        var name = sys.name(src);
+        var signupsLower = this.signups.map(function(x) { return x.toLowerCase(); });
+        if (signupsLower.contains(name.toLowerCase())) {
+            safaribot.sendMessage(src, "You already joined the event!", safchan);
+            return;
+        }
+        if (player.party.length < 6) {
+            safaribot.sendMessage(src, "You must have a party with 6 Pokémon to join this event!", safchan);
+            return;
+        }
+        
+        this.sendToViewers(name + " has joined the " + this.eventName + "!");
+        this.signups.push(name);
+        
+        safaribot.sendMessage(src, "You joined the " + this.eventName + "!", safchan);
+    };
+    FactionWar.prototype.watchEvent = function(src) {
+        var name = sys.name(src);
+        var signupsLower = this.signups.map(function(x) { return x.toLowerCase(); });
+        if (signupsLower.contains(name.toLowerCase())) {
+            safaribot.sendMessage(src, "You are already in the event!", safchan);
+            return;
+        }
+        
+        if (this.viewers.contains(name.toLowerCase())) {
+            this.sendToViewers(name + " stopped watching this " + this.eventName + "!");
+            this.viewers.splice(this.viewers.indexOf(name.toLowerCase()), 1);
+            safaribot.sendMessage(src, "You are no longer watching the " + this.eventName + "!", safchan);
+        } else {
+            this.sendToViewers(name + " is watching this " + this.eventName + "!");
+            this.viewers.push(name.toLowerCase());
+            safaribot.sendMessage(src, "You are watching the " + this.eventName + "!", safchan);
+        }
+    };
+    FactionWar.prototype.sendMessage = function(name, msg) {
+        var id = sys.id(name);
+        if (id) {
+            if (msg === "") {
+                sys.sendHtmlMessage(id, msg, safchan);
+            } else {
+                safaribot.sendHtmlMessage(id, msg, safchan);
+            }
+        }
+    };
+    FactionWar.prototype.sendToViewers = function(msg) {
+        var e;
+        var list = this.signups.concat(this.viewers);
+        for (e in list) {
```

Comment:
The body of a for in should be wrapped in an if statement to filter unwanted properties from the prototype.